Show fake data

Naomi-Dennis opened the pull request.

Naomi-Dennis committed:
Remove unnecessary test

Naomi-Dennis committed:
Add Index homepage displaying fake data in table

Naomi-Dennis committed:
Separate view and model definitions from Main

damonkelley commented on src/views/helpers/IndexHelpers.elm:
```diff
@@ -0,0 +1,14 @@
+module IndexHelpers exposing (..)
```
What is `IndexHelpers`?

Naomi-Dennis replied:
They're functions that help support the view it's named after. The `Index.elm` file is being treated as an `HTML` only file. So it only has Html literals. I don't want to distract the view from the overall structure of the webpage.

damonkelley replied:
I am always wary of "helpers" or "utility" because they are a catch-all and can become meaningless.

damonkelley commented on src/Models.elm:
```diff
@@ -0,0 +1,9 @@
+module Models exposing (..)
```
Is there a name that would map better to the domain of the application vs the content of the file?

Naomi-Dennis replied:
I know! `Models` is a temporary name, I'm looking into changing into something more accurate.

damonkelley commented on src/views/Index.elm:
```diff
@@ -0,0 +1,20 @@
+module Index exposing (..)
```
Can you elaborate on this name?

Naomi-Dennis replied:
`Index` refers to the homepage (in this case the only page) of the site. Index refers to `www.domain.com/` or `www.domain.com/index`. 

It's not a naming convention of elm, I just thought it was descriptive of where the view would be shown

damonkelley replied:
Gotcha, I was wondering if maybe where it is shown would be orthogonal to what it is doing. I don't really like `Index` all that much because it is in the domain of HTTP convention, vs the domain of this application.

Naomi-Dennis replied:
I agree. It's now `Homepage`

damonkelley commented:
It looks like there is a commit that are perhaps already in master and an a merge commit. 

1591f08 and 3d98a93

Can we remove these from the PR?

Naomi-Dennis committed:
Fix travis only deploy on master

Travis wasn't adhering to the `if: branch = master`
condition since the base branch is master. An additional
condition should fix that problem and only deploy the application
on the master branch.

hnlee commented on src/ArrivalTypes.elm:
```diff
@@ -0,0 +1,9 @@
+
+
+type Msg
+    = ShowArrivalTimes
```
Noticing that elsewhere you're putting `=` on the same line as the type being defined, but here you've put it on the next line. Is this just some obscure Elm convention I'm missing? I know there's supposed to be a linter that comes with the language...

Naomi-Dennis replied:
It's done by the linter. I think when defining a type that isn't an alias this happens.

damonkelley commented on tests/IndexHelpersTest.elm:
```diff
@@ -0,0 +1,51 @@
+module IndexHelpersTest exposing (..)
```
Is this name out of date?

Naomi-Dennis replied:
Ty for the catch!

damonkelley commented on src/views/helpers/HomepageHelper.elm:
```diff
@@ -0,0 +1,14 @@
+module HomepageHelper exposing (..)
```
I would probably opt to include this the `Homepage` module until it made sense to break it out into a new module that makes sense.

damonkelley commented on src/views/Homepage.elm:
```diff
@@ -0,0 +1,20 @@
+import ArrivalTypes exposing (Msg, StationArrivalTimes)
+
+
+index : StationArrivalTimes -> Html Msg
```
What do you think about this name? I would opt for something a bit more descriptive. 

You could leverage the type signature in the name too, so something like this maybe?

```elm
show : StationArrivalTimes -> Html Msg
show stationArrivalTimes =
  // ..
```

So that would read as *show station arrival times*

damonkelley commented on src/views/Homepage.elm:
```diff
@@ -0,0 +1,20 @@
+module Homepage exposing (..)
```
I am not sold on this but I do think it is more fitting than `Index` 😸
